src/app/SmartR/components/services/rServeService.js

```diff
@@ -63,6 +63,46 @@ angular.module('smartRApp').factory('rServeService', [
             });
         };
 
+        service.fetchImageResource = function(uri) {
+            var authHeaders = EndpointService.getMasterEndpoint().restangular.defaultHeaders;
+            authorizationHeader = authHeaders['Authorization'];
+
+            var deferred = $q.defer();
+
+            var xmlHTTP = new XMLHttpRequest();
+            xmlHTTP.open('GET',uri, true);
+            xmlHTTP.responseType = 'arraybuffer';
+            xmlHTTP.setRequestHeader("Authorization", authorizationHeader);
+
+            xmlHTTP.onload = function() {
+                var arr = new Uint8Array(this.response);
+
+                // Convert the int array to a binary string
+                // We have to use apply() as we are converting an *array*
+                // and String.fromCharCode() takes one or more single values, not
+                // an array.
+                var raw = String.fromCharCode.apply(null,arr);
+                var b64=btoa(raw);
```

Comment:
A hint: if you use cmd + alt +  L in intellij on mac (with default keyboard settings) it will format the code nicely for you. So e.g. you don't have to worry about missing white-spaces between operators, it also fixes most of indentation issues!